### ISSUE TITLE
Parser hangs on parsing tables

### ISSUE THREAD

**Issue by krmrn42** [CXuesong/MwParserFromScratch#12]
The parser hangs on parsing the table in Noun section of the Wiktionary page [depth](https://en.wiktionary.org/wiki/depth). Stripping table from the content of this section makes it pass.

**Comment by CXuesong** [comment]
There seems to be too many fallbacks at the closing `</table>`.

I fed the following text to ParserTracer
```wikitext
<TABLE>
<TR><TD></TD><TD COLSPAN="9">Ordered Batch of 9 Values</TD></TR>
</TABLE>
```

The output is as follows
```
Parsed 85 characters in 0.0394467 sec.
Regex Matching
    Count  Elpsd ms   Average ms Expression
       170     0.8493     0.0049 \n
         8       0.03     0.0037 /?>|[\s=]
        30      0.065     0.0021 </(table)(\s*)>
        28     0.0558     0.0019 </(tr)(\s*)>
        44     0.0629     0.0014 </(td)(\s*)>
         8     0.0117     0.0014 [>"]|/>

Fallbacks: 405
    Count  Position Text
       112     75 </TABLE>\r\n
        48     26 COLSPAN="9">Ordered Batch of 9
        48     35 9">Ordered Batch of 9 Values</
        48     38 Ordered Batch of 9 Values</TD>
        40     22 <TD COLSPAN="9">Ordered Batch
        20     13 <TD></TD><TD COLSPAN="9">Order
        20     17 </TD><TD COLSPAN="9">Ordered B
        16     76 /TABLE>\r\n
        15     85
        14      9 <TR><TD></TD><TD COLSPAN="9">O
         8     34 "9">Ordered Batch of 9 Values<
         6      0 <TABLE>\r\n<TR><TD></TD><TD COLS
         6      7 \r\n<TR><TD></TD><TD COLSPAN="9"
         4     18 /TD><TD COLSPAN="9">Ordered Ba
Parsed tree
.Wikitext            [<TABLE>\r\n<TR><TD></T]
..Paragraph          [<TABLE>\r\n<TR><TD></T]
...PlainText         [<TABLE>\r\n<TR><TD></T]
```

Somehow the parser didn't recognize the TABLE element.

**Comment by CXuesong** [comment]
Actually, the parser failed to recognize all of the HTML tags that is not in lower-case. e.g. `<TABLE>test</TABLE>` is parsed as plain text.

**Comment by CXuesong** [comment]
Fixed in release [`0.2.1`](https://github.com/CXuesong/MwParserFromScratch/releases/tag/0.2.1). Note there are also some API adjustments involved for making wikitext table parsing possible in the future. (Ugh I should have created a branch and worked on the branch) If this causes trouble to you, please leave me a message here. Thanks.
